### DISCUSSION TITLE
How to choose .net core version?

### DISCUSSION THREAD

**Discussion by sridharbaldava** [dotnet/interactive#1375] [Q&A] [answered]
Is there a way to force Jupyter Notebook to use a particular version of .net? It seems to be defaulting to .Net Core 3.1 even though I have .net 5.0 installed.

.net-interactive-csharp\2e536de046f04d5a8183344a9ac84db2\r.csproj : error NU1202: Package Microsoft.DotNet.Interactive.ExtensionLab 1.0.0-beta.21228.1 is not compatible with netcoreapp3.1 (.NETCoreApp,Version=v3.1). Package Microsoft.DotNet.Interactive.ExtensionLab 1.0.0-beta.21228.1 supports: net5.0 (.NETCoreApp,Version=v5.0)

**Comment by jonsequitur** [comment, marked as answer]
It looks like it's defaulting to .NET 5. We don't currently have a way to choose the version.

**Reply by sridharbaldava** [reply]
Looking at the temporary csproj at the location above it is selecting netcoreapp3.1
```
<PropertyGroup>
        <OutputType>Exe</OutputType>
        <TargetFramework>netcoreapp3.1</TargetFramework>
        <IsPackable>false</IsPackable>
</PropertyGroup>
```

When running the same notebook via VSCode it is defaulting to net5 app and working as expected. I will continue to test jupyter some more.

**Reply by jonsequitur** [reply]
What version of .NET Interactive do you have installed globally? You can check by running `dotnet interactive --version` in your terminal.

**Reply by sridharbaldava** [reply]
1.0.115407+d247e88f3f756abad4bd16b47cf8cde187aafabc

**Reply by jonsequitur** [reply]
Ah, I see. The version you're using is a bit older and targets .NET Core 3.1. We shifted to .NET 5 a number of months back.

```console
dotnet tool update -g Microsoft.dotnet-interactive
```

**Reply by sridharbaldava** [reply]
I just updated and can report that it now works as expected. Thanks again for your help with resolving this.

**Reply by jonsequitur** [reply]
You're welcome!
